src/hoi4_world/states/hoi4_states_converter.cpp

```diff
@@ -284,17 +352,36 @@ hoi4::States CreateStates(const std::map<int, vic3::State>& vic3_states,
          }
          vic3_state_ids_to_hoi4_state_ids.emplace(vic3_state_id, static_cast<int>(hoi4_states.size() + 1U));
          hoi4_states.emplace_back(static_cast<int>(hoi4_states.size() + 1U),
-             hoi4::StateOptions{.owner = state_owner, .provinces = province_set, .manpower = manpower});
+             hoi4::StateOptions{.owner = state_owner,
+                 .provinces = province_set,
+                 .manpower = manpower,
+                 .civilian_factories = civilian_factories,
+                 .military_factories = military_factories,
+                 .dockyards = dockyards});
       }
    }
 
+   int civilian_factories = 0;
+   int military_factories = 0;
+   int dockyards = 0;
+   for (const auto& hoi4_state: hoi4_states)
+   {
+      civilian_factories += hoi4_state.GetCivilianFactories();
+      military_factories += hoi4_state.GetMilitaryFactories();
+      dockyards += hoi4_state.GetDockyards();
+   }
+
+   Log(LogLevel::Info) << fmt::format("\t\tTotal factories: {}", civilian_factories + military_factories + dockyards);
```

Comment:
You logging this for every state? That'll be a hefty log.

Reply:
The huge function makes it unclear, but its only once after everything.